internal/builder/main.go

```diff
@@ -105,6 +105,39 @@ func Compile(cfg Config) error {
 	return nil
 }
 
+// GetModules Runs go mod download to update go.mod and go.sum
+func GetModules(cfg Config) error {
+	// first, we test to check if we have Go at all
+	goBinary, err := GetGoPath(cfg)
+	if err != nil {
+		return err
+	}
+
+	cfg.Logger.Info("Getting go modules")
+	cmd := exec.Command(goBinary, "mod", "download")
+	cmd.Dir = cfg.Distribution.OutputPath
+	if out, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("failed to get go modules: %w. Output: %q", err, out)
```

Comment:
```suggestion
		return fmt.Errorf("failed to download go modules: %w. Output: %q", err, out)
```

Reply:
Is it possible to determine whether the failure was permanent? Does it make sense to retry, especially on network connectivity issues?

Reply:
I'll look if `go mod download` gives information to the type of the error (network, transient etc) and how easy it can be parsed. If go mod gives this information I could then retry but wonder how many times/ how long it should retry for. For example we run this in a build pipeline which has a timeout. If it took too long to retry it would cause the pipeline to timeout.

Reply:
To make it easier to get this fix ready for 1.16, could you create an issue tracking this idea? Or perhaps even simpler: try at most 3 times before failing permanently.

Reply:
Added basic retry. Will create an issue to improve at a later stage